lib/exgboost.ex

```diff
@@ -204,69 +171,71 @@ defmodule EXGBoost do
+      to `true` is not recommended.
 
-  * `:pred_interactions` - When this is `true` the output will be a matrix of size (nsample,
-      nfeats + 1, nfeats + 1) indicating the SHAP interaction values for
+  * `:pred_interactions` - When this is `true` the output will be an `Nx.Tensor` of shape
+      {nsamples, nfeats + 1} indicating the SHAP interaction values for
       each pair of features. The sum of each row (or column) of the
       interaction values equals the corresponding SHAP value (from
       pred_contribs), and the sum of the entire matrix equals the raw
       untransformed margin value of the prediction. Note the last row and
       column correspond to the bias term.
 
   * `:validate_features` - When this is `true`, validate that the Booster's and data's
-      feature_names are identical.  Otherwise, it is assumed that the
+      feature_names are identical. Otherwise, it is assumed that the
       feature_names are the same.
 
-  * `:training` - Whether the prediction value is used for training.  This can effect `dart`
-      booster, which performs dropouts during training iterations but use all trees
-      for inference. If you want to obtain result with dropouts, set this parameter
-      to `true`.  Also, the parameter is set to `true` when obtaining prediction for
+  * `:training` - Determines whether the prediction value is used for training. This
+      can affect the `dart` booster, which performs dropouts during training iterations
+      but uses all trees for inference. If you want to obtain result with dropouts, set
+      this option to `true`. Also, the option is set to `true` when obtaining prediction for
       custom objective function.
 
-  * `:iteration_range` - Specifies which layer of trees are used in prediction.  For example, if a
-      random forest is trained with 100 rounds.  Specifying `iteration_range=(10,
+  * `:iteration_range` - Specifies which layer of trees are used in prediction. For example, if a
+      random forest is trained with 100 rounds. Specifying `iteration_range=(10,
       20)`, then only the forests built during [10, 20) (half open set) rounds are
       used in this prediction.
 
   * `:strict_shape` - When set to `true`, output shape is invariant to whether classification is used.
       For both value and margin prediction, the output shape is (n_samples,
-      n_groups), n_groups == 1 when multi-class is not used.  Default to False, in
+      n_groups), n_groups == 1 when multi-class is not used. Defaults to `false`, in
       which case the output shape can be (n_samples, ) if multi-class is not used.
 
   Returns an Nx.Tensor containing the predictions.
   """
   def predict(%Booster{} = bst, %Nx.Tensor{} = x, opts \\ []) do
     {dmat_opts, opts} = Keyword.split(opts, Internal.dmatrix_feature_opts())
-    dmat = EXGBoost.DMatrix.from_tensor(x, [format: :dense] ++ dmat_opts)
+    dmat = DMatrix.from_tensor(x, Keyword.put_new(dmat_opts, :format, :dense))
```

Comment:
I think put_new makes sense. The :format key is really sort of something I had in case sparse tensors became a thing. The whole XGBoost API lends itself more to sparse matrices, but I dont think sparse tensors exist yet in Nx. For now, :format isn't even needed